graphene_django/utils/utils.py

```diff
@@ -145,3 +145,18 @@ def bypass_get_queryset(resolver):
     """
     resolver._bypass_get_queryset = True
     return resolver
+
+
+def __django_version():
+    from pkg_resources import get_distribution
```

Comment:
There's a warning at the top here https://setuptools.pypa.io/en/latest/pkg_resources.html saying

> Use of pkg_resources is deprecated in favor of [importlib.resources](https://docs.python.org/3/library/importlib.resources.html#module-importlib.resources), [importlib.metadata](https://docs.python.org/3/library/importlib.metadata.html#module-importlib.metadata) and their backports ([importlib_resources](https://pypi.org/project/importlib_resources), [importlib_metadata](https://pypi.org/project/importlib_metadata)). Some useful APIs are also provided by [packaging](https://pypi.org/project/packaging) (e.g. requirements and version parsing). Users should refrain from new usage of pkg_resources and should work to port to importlib-based solutions.

Instead of `parse_version()`, it looks like we could use https://packaging.pypa.io/en/stable/version.html (`from packaging.version import parse`). And for getting the currently-installed django version string, it looks like `import django; django.get_version()` would work (or alternatively [`importlib.metadata.version`](https://docs.python.org/3/library/importlib.metadata.html#module-importlib.metadata) seems to be the substitute for `get_distribution("...").parsed_version`).

Unsure how important this deprecation warning is, but seems like it may be worthwhile to avoid the new code depending on this, as they suggest.

---

Another alternative could be something like this (if we want to avoid `packaging` for instance):

```python
import django

_DJANGO_VERSION_AT_LEAST_4_2 = django.VERSION[0] > 4 or (django.VERSION[0] >= 4 and django.VERSION[1] >= 2)
``` 
```

Reply:
@sjdemartini @superlevure Thanks for the review. I know about `pkg_resources` being deprecated in favor of `importlib...` modules. However went for `pkg_resources.get_distribution("...").parsed_version` here since we need to compare semver while `importlib.metadata.version` only returns the version string, unparsed.

`django.VERSION` looks like exactly what we need which I was not aware of. Will convert to that. Thanks for pointing that out!